src/combinatorics/simple_graph/basic.lean

```diff
@@ -332,4 +332,59 @@ finset.max' (univ.image (λ (v : V), G.degree v)) (nonempty.image univ_nonempty
 
 end finite
 
+section complement
+
+/-!
+## Complement of a simple graph
+
+This section contains definitions and lemmas concerning the complement of a simple graph.
+-/
+
+/--
+We define `compl G` to be the `simple_graph V` such that no two adjacent vertices in `G`
+are adjacent in the complement, and every nonadjacent pair of vertices is adjacent
+(still ensuring that vertices are not adjacent to themselves.)
+-/
+@[simps]
```

Comment:
To fix your problem, remove `@[simps]` and then add this after `has_compl`:
```lean
@[simp]
lemma compl_adj (G : simple_graph V) (v w : V) : Gᶜ.adj v w ↔ v ≠ w ∧ ¬G.adj v w := iff.rfl
```
This is the problem with notation: you need to make sure all lemma statements use it for them to be able to do rewrites.

Reply:
Gotcha, thanks!

Reply:
I assumed Lean would pick up on the fact that the notation refers to `compl` so it was weird when it didn't

Reply:
@kmill Would it be better instead to add:
```lean
@[simp] lemma compl_def (G : simple_graph V) : Gᶜ = G.compl := rfl
```

This would let us `simp` or `rw` away the notation more generally.

I think we could keep `@[simps]` in that case, right?

Reply:
I think that would mean that every lemma would then have to be in terms of `G.compl` to be applicable by `simp` and `rw`.  It seems to be that writing out `compl_adj` is the cost to using fancy notation.

(If anything, that lemma, if it were to be included, should be the other way so that every instance of `G.compl` that might show up can be easily turned into the fancy `has_compl.compl G`.)

Reply:
Ah, I see. Thanks for the explanation!